### PR TITLE
feat(quantic): disclaimer message for rga

### PR DESCRIPTION
**Pull request by erocheleau**
Added a disclaimer at the bottom of the RGA message.

The disclaimer has a default look and text of : 

`Generated content may contain errors. Verify important information.`

And the section for the disclaimer is also a slot, so if someone wants to completely change what gets displayed as the disclaimer, they can not only edit the text, but also the HTML.

The message is also a label, so it can be translated, etc.

Single line footer:
image

Multi-line footer:
image

In the Insight Panel:
image

### CODE REVIEW

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/// <reference types="cypress" />
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:1]: is this fixing the typing issues?

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 2]
```diff
@@ -136,6 +140,12 @@ describe('quantic-generated-answer', () => {
             Expect.generatedAnswerIsStreaming(false);
           });
 
+          it('should display the disclaimer', () => {
+            Expect.displayDisclaimer(true);
+            Expect.disclaimerContains(GENERATED_ANSWER_DISCLAIMER);
+            Expect.generatedAnswerIsStreaming(false);
+          });
+
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:148]: We should add the same test to the describe `describe('when the property multilineFooter is set to true', ...)` which can be found on line 215.

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 3]
```diff
@@ -1400,11 +1400,18 @@
...
   </labels>
-    <labels>
+  <labels>
     <fullName>quantic_GeneratedAnswerOff</fullName>
     <value>Generated answer OFF</value>
     <language>en_US</language>
     <protected>false</protected>
     <shortDescription>Generated answer OFF</shortDescription>
   </labels>
+  <labels>
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:1410]: Please also include the french translation.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js [hunk 4]
```diff
@@ -84,6 +85,13 @@ export default class QuanticGeneratedAnswer extends LightningElement {
    * @default {false}
    */
   @api multilineFooter;
+  /**
+   * Indicates whether the disclaimer should be hidden.
+   * @api
+   * @type {boolean}
+   * @default {false}
+   */
+  @api noDisclaimer;
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:94]: Just my opinion but I feel that naming the property `withoutDisclaimer` or `hideDisclaimer` feels a bit more natural.

Plus if we keep the public property it will need to be tested with Cypress as well.

Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:94]: Also is really necessary to have this property, like we are already exposing a slot for the disclaimer so we already have a way to get a generated answer component without a disclaimer.

I assume that the majority of the people using this component would want the disclaimer since it's useful and they can customize its content.
for the few/rare users who don't want a disclaimer, they can remove it but specifying an empty slot, I know it would not be a super fancy way  to hide a disclaimer, but considering that only very few people will need to do that I would not add a new public property just for this particular situation. 

The idea is I don't see the point of having a public property that allows you to do something that you can achieve without using this property.

**Review by mmitiche — COMMENTED** [review]
The disclaimer looks good!

Couple adjustments to the tests, french translations, and a question raised about the necessity of a new public property.

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts (hunk 1 above)
Comment (erocheleau) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:1]: woops, no it wasn't working, forgot to remove it.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 4 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:94]: Do you know if we can empty a slot? If we can then yes, I wouldn't have this property

Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:94]: This should do it:
```
   <c-quantic-generated-answer
      engine-id={engineId}
      answer-style={answerStyle}
      multiline-footer={multilineFooter}
    >
      <div slot="disclaimer"></div>
   </c-quantic-generated-answer>
```

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.html [hunk 5]
```diff
@@ -65,6 +65,13 @@
                 ></c-quantic-generated-answer-rephrase-buttons>
               </div>
             </template>
+            <template lwc:if={shouldShowDisclaimer}>
+              <slot name="disclaimer">
+                <div class={disclaimerCssClass} data-cy="generated-answer__disclaimer">
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.html:70]: Some of the styling classes present in `disclaimerCssClass ` needs to be outside the slot, otherwise all these styles will not be take in consideration when displaying custom disclaimers yielding an unwanted display.

For example:
```
<c-quantic-generated-answer engine-id={engineId} answer-style={answerStyle} multiline-footer={multilineFooter}>
    <div slot="disclaimer">Custom Disclaimer</div>
</c-quantic-generated-answer>
```

will give me this::
 
Screenshot 2024-03-22 at 9 28 27 AM

File: packages/quantic/quantic-examples-community/experiences/Quantic_Examples1/views/insightPanelExample.json [hunk 6]
```diff
@@ -34,7 +34,7 @@
                 {
                   "componentAttributes": {
                     "engineId": "example-insight-panel",
-                    "insightId": "6a333202-b1e0-451e-8664-26a1f93c2faf"
+                    "insightId": "142be676-703c-445f-b2d3-fcc7c0a3ded8"
```
Comment (SimonMilord) [inline packages/quantic/quantic-examples-community/experiences/Quantic_Examples1/views/insightPanelExample.json:37]: why this change? did we change the config or something?

**Review by SimonMilord — APPROVED** [review]
Mehdi's comment pretty much cover it imo, otherwise looks good!

File: packages/quantic/quantic-examples-community/experiences/Quantic_Examples1/views/insightPanelExample.json (hunk 6 above)
Comment (mmitiche) [reply packages/quantic/quantic-examples-community/experiences/Quantic_Examples1/views/insightPanelExample.json:37]: Since we are it, let's also remove/update where this configuration id is set in other places, it is set in multiple places and it is not always useful: 

-  https://github.com/coveo/ui-kit/blob/8f4d68d7a54d6aa539ed85b7fd0516c123097eb7/packages/quantic/force-app/examples/main/lwc/exampleInsightPanel/exampleInsightPanel.js-meta.xml#L15
- https://github.com/coveo/ui-kit/blob/8f4d68d7a54d6aa539ed85b7fd0516c123097eb7/packages/quantic/force-app/examples/main/lwc/exampleInsightPanel/exampleInsightPanel.js#L24

File: packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts [hunk 7]
```diff
@@ -258,6 +268,7 @@ describe('quantic-generated-answer', () => {
             Expect.displayCopyToClipboardButton(false);
             Expect.displayToggleGeneratedAnswerButton(true);
             Expect.toggleGeneratedAnswerButtonIsChecked(true);
+            Expect.displayDisclaimer(false);
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:271]: The test already checks that the `generated-answer__card` isn't being displayed, the disclaimer is inside the answer card, so if the card isn't displayed, then the disclaimer isn't displayed either.

[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts:271]: But I added it after the answer card is back to visible.

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js (hunk 4 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js:94]: ok then I'm happy to remove the option :)

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.html (hunk 5 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.html:70]: Yes my goal with not putting the css class outside the slot was to allow the customers to style their own disclaimer without "competing" with our own style.

I know it's easier for them if they just want to add some text, but if they would want to display the disclaimer differently it can become impossible if the classes are outside the slot. What do you think?

File: packages/quantic/quantic-examples-community/experiences/Quantic_Examples1/views/insightPanelExample.json (hunk 6 above)
Comment (erocheleau) [reply packages/quantic/quantic-examples-community/experiences/Quantic_Examples1/views/insightPanelExample.json:37]: Yea sure we can remove it in the default example component.

@SimonMilord This is just a different config for the HIP which has RGA enabled and is using a query we know will generate a RGA response.

File: packages/quantic/force-app/examples/main/lwc/exampleInsightPanel/exampleInsightPanel.js-meta.xml [hunk 8]
```diff
@@ -12,7 +12,7 @@
     <targetConfigs>
         <targetConfig targets="lightningCommunity__Default, lightning__AppPage, lightning__RecordPage, lightning__HomePage">
             <property name="engineId" type="String" default="example-insight-panel" label="Enter Coveo Headless engine ID"/>
-            <property name="insightId" type="String" default="6a333202-b1e0-451e-8664-26a1f93c2faf" label="The insight panel config ID"/>
+            <property name="insightId" type="String" default="" label="The insight panel config ID"/>
```
Comment (mmitiche) [inline packages/quantic/force-app/examples/main/lwc/exampleInsightPanel/exampleInsightPanel.js-meta.xml:15]: ?
```suggestion
            <property name="insightId" type="String" label="The insight panel config ID"/>
```

File: packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.html (hunk 5 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.html]: Indeed I get your point, but we can at least set the class that would display the disclaimer  under the citation and the rephrase buttons outside the slot so we know for sure that the disclaimer will be positioned in the right position.
Then, all the other classes can be set inside the slot so they can easily be overwritten by the classes that the user wanna use to style the disclaimer.

**Review by mmitiche — APPROVED** [review]
Looks good to me 👍

File: packages/quantic/force-app/examples/main/lwc/exampleInsightPanel/exampleInsightPanel.js-meta.xml (hunk 8 above)
Comment (mmitiche) [reply packages/quantic/force-app/examples/main/lwc/exampleInsightPanel/exampleInsightPanel.js-meta.xml:15]: Sorry I didn't explain well my comment, I meant why not completely remove the `default=""` since it does nothing.
